src/main/java/se/lth/base/server/mail/CustomEmail.java

```diff
@@ -2,64 +2,510 @@
+        content.put(BUTTON, getButton(WEBSITE_LINK, "Get Started Here"));
+
+        // Add recipient
+        String fullName = singleUser.getFirstName() + " " + singleUser.getLastName();
+        recipients.add(new Recipient(fullName, singleUser.getEmail(), Message.RecipientType.TO));
+
+        return parseResource(getResource(EMAIL_TEMPLATE), content);
+    }
+
+    private String getNewPassengerOnTripSubject() {
+        return "New passenger request for your drive to " + driveWrap.getDrive().getStop();
+    }
+
+    private String getNewPassengerOnTripBody() {
+        Map<String, String> content = new HashMap<>();
+        DriveUser lastPassenger = null, driver = null;
+        Timestamp departureTime = driveWrap.getDrive().getDepartureTime();
+        String formattedDepartureTime = new SimpleDateFormat("EEEEE MMMMM d 'at' HH:mm").format(departureTime);
+
+        // Fetch the driver and last passenger added
+        for (DriveUser u : driveWrap.getUsers()) {
+            if (u.isDriver()) {
+                driver = u;
+            }
+
+            lastPassenger = u;
+        }
+
+        // Get the user data
+        UserDataAccess userDao = new UserDataAccess(Config.instance().getDatabaseDriver());
+        assert driver != null;
+        User driverU = userDao.getUser(driver.getUserId());
+
+        // Title
+        content.put(TITLE, getNewPassengerOnTripSubject());
+
+        // Intro
+        String intro = "<p>Hi " + driverU.getFirstName() + ",</p>";
+        intro += "<p>A passenger has requested to travel with you from <i>";
+        intro += lastPassenger.getStart() + "</i> to <i>" + lastPassenger.getStop();
+        intro += "</i> on your drive from <i>" + driver.getStart() + "</i> to <i>";
+        intro += driver.getStop() + "</i> on " + formattedDepartureTime + ".</p>";
+        content.put(INTRO, intro);
+
+        // Action button
+        content.put(BUTTON, getButton(WEBSITE_LINK + "/drive/" + driveWrap.getDrive().getDriveId(), "View Drive"));
+
+        // Add recipient
+        String fullName = driverU.getFirstName() + " " + driverU.getLastName();
+        recipients.add(new Recipient(fullName, driverU.getEmail(), Message.RecipientType.TO));
+
+        return parseResource(getResource(EMAIL_TEMPLATE), content);
+    }
+
+    private String getBookingConfirmedSubject() {
+        return "Your booking request is confirmed";
+    }
+
+    private String getBookingConfirmedBody() {
+        Map<String, String> content = new HashMap<>();
+        DriveUser acceptedPassenger = null, driver = null;
+
+        // Fetch the driver and last passenger added
+        for (DriveUser u : driveWrap.getUsers()) {
+            if (u.isDriver()) {
+                driver = u;
+            }
+
+            acceptedPassenger = u;
+        }
+
+        Timestamp departureTime = driveWrap.getDrive().getDepartureTime();
+        String formattedDepartureTime = new SimpleDateFormat("EEEEE MMMMM d").format(departureTime);
```

Comment:
Some `departureTime` are formatted to include time as well. Others just include date, like this one. If you only want to present date; use `formattedDepartureDate` for clarity.